packages/blazor-workspace/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js

```diff
@@ -162,12 +163,48 @@ export function afterStarted(Blazor) {
     });
 }
 
+function handleRuntimeStarted() {
+    window.NimbleBlazor.hasRuntimeStarted = true;
+}
+
+// Blazor Web Apps
+export function afterWebStarted(Blazor) {
+    registerNimbleEvents(Blazor);
+    // Note: For static SSR, this is the last event called, and hasRuntimeStarted
+    // will remain false.
+}
+
+export function afterServerStarted(_Blazor) {
+    handleRuntimeStarted();
+}
+
+// Blazor Web Apps, WASM Standalone apps with .NET 8
+export function afterWebAssemblyStarted(_Blazor) {
+    if (!window.NimbleBlazor.hasRegisteredEvents) {
+        registerNimbleEvents(Blazor);
+    }
+
+    handleRuntimeStarted();
+}
+
+// Blazor Server/WebAssembly/Hybrid apps
+export function afterStarted(Blazor) {
+    // In some cases afterStarted is called on Blazor Web Apps too, if Nimble is used in a component explicitly
+    // marked as InteractiveWebAssembly render mode. So check if we've already registered our events first.
+    if (!window.NimbleBlazor.hasRegisteredEvents) {
+        registerNimbleEvents(Blazor);
+    }
+
+    handleRuntimeStarted();
+}
+
 if (window.NimbleBlazor) {
     console.warn('Attempting to initialize NimbleBlazor multiple times!'); // eslint-disable-line
 }
 
 window.NimbleBlazor = window.NimbleBlazor ?? {
-    calledAfterStarted: false,
+    hasRegisteredEvents: false,
+    hasRuntimeStarted: false,
```

Comment:
Do we actually need to expose these as global state?
Maybe we can have a NimbleBlazor.waitReady Promise instead that we can await on from tests.

```
let resolveReady;
const ready = new Promise(resolve => {resolveReady = resolve}); 
// ...
afterWebStarted() { resolveReady();}
// ...
window.NimbleBlazor = {
    waitReady: async () => await ready
}
```

Reply:
Thought about this more, and don't think it works as-is. There's no guarantee that the Nimble/Spright JS initialization scripts have loaded by the time that the Playwright browser (acceptance tests) has loaded the page and is calling the JS to determine if we're initialized or not. (More simply, window.NimbleBlazor will be undefined until that JS script has loaded, so invoking `window.NimbleBlazor.waitReady()` would throw an error before that point.)

I did revamp the script so that the boolean `hasRegisteredEvents` flag is no longer global / part of the `window.NimbleBlazor` object. A single `isInitialized` flag is exposed for the C# tests to query for instead.